### PR TITLE
Added Improved QuickSort.cpp

### PR DESCRIPTION
**Pull request by suraj0223**
Implementation of 3 way partition Quick Sort.

### CODE REVIEW

**Comment by diptangsu** [comment]
Just rename the file to `ThreeWayQuickSort.cpp`

**Comment by suraj0223** [comment]
@diptangsu Please Review again Updates done.

**Comment by diptangsu** [comment]
Is there a reason why the partition function begins with a capital P?
Also, it would be fair to call the function `threeWayQuickSort` instead of whatever snake case naming convention you've used.

**Comment by suraj0223** [comment]
3 way partition Quick sort perform better when there is repeated element in the list as compared to 2 way partition Quick sort.
check it out here : [ 3 way sorting visualization](https://www.toptal.com/developers/sorting-algorithms)

File: Cpp/ThreeWayQuickSort.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
+// This is a implementation of 3 way partition QuickSort.
+#include<bits/stdc++.h>
+using namespace std;
+
+pair<int,int> Partition(vector<int> &v1, int l, int r) {
```
Comment (diptangsu) [inline Cpp/ThreeWayQuickSort.cpp:5]: refactor to `partition`

File: Cpp/ThreeWayQuickSort.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     else if (v1[mid] > pivot)
+        swap(v1[mid], v1[r--]);
+  }
+  i = l-1;
+  j = mid;
+
+  return make_pair(i,j);
+}
+
+void randomized_quick_sort(vector<int> &v1, int l, int r) {
```
Comment (diptangsu) [inline Cpp/ThreeWayQuickSort.cpp:30]: change all snake case function names to camelCase names

**Review by diptangsu — CHANGES_REQUESTED** [review]
Please make the necessary changes.

**Review by diptangsu — APPROVED** [review]

**Pull request merged**